crates/nu-cli/src/shell/filesystem_shell.rs

```diff
@@ -1000,125 +1002,143 @@ impl Shell for FilesystemShell {
-                                                    ))
-                                                },
-                                            }
-                                        }
-                                    } else {
-                                        if f.is_dir() {
-                                            let msg = format!(
-                                                "Cannot remove {:}. try --recursive",
-                                                f.to_string_lossy()
-                                            );
-                                            yield Err(ShellError::labeled_error(
-                                                msg,
-                                                "cannot remove non-empty directory",
-                                                &target.tag,
-                                            ))
-                                        } else {
-                                            let msg = format!("Invalid file: {:}", f.to_string_lossy());
-                                            yield Err(ShellError::labeled_error(
-                                                msg,
-                                                "invalid file",
-                                                &target.tag,
-                                            ))
-                                        }
-                                    }
+            let mut path = PathBuf::from(path);
+            path.push(&target.item);
+            match glob::glob(&path.to_string_lossy()) {
+                Ok(files) => {
+                    for file in files {
+                        match file {
+                            Ok(ref f) => {
+                                if !unique_paths.contains(f) {
+                                    unique_paths.insert(f.clone());
+                                    all_targets.push(Tagged::new(target.tag.clone(), f.clone()));
                                 }
+                            }
+                            Err(e) => {
+                                let msg = format!("Could not remove {:}", path.to_string_lossy());
+                                return Err(ShellError::labeled_error(
+                                    msg,
+                                    e.to_string(),
+                                    &target.tag,
+                                ));
+                            }
+                        }
+                    }
+                }
+                Err(e) => {
+                    return Err(ShellError::labeled_error(
+                        format!("Remove aborted. {:}", e.to_string()),
+                        e.to_string(),
+                        &name_tag,
+                    ))
+                }
+            };
+        }
+
+        if all_targets.is_empty() {
+            Err(ShellError::labeled_error(
+                "Remove aborted. Not a valid path",
```

Comment:
Let's change the message here too. Since we potentially have more than a single path, maybe we should have something like this:

```rust
ShellError::labeled_error(
  "Remove aborted. No valid paths.",
  "no valid paths",
  name_tag
)
```

Reply:
Thanks!